composables/useNFTContractWriter.ts

```diff
@@ -31,14 +32,26 @@ export const useNFTContractWriter = () => {
       console.log(`Granted ${role} to ${wallet} in class ${classId}. Transaction hash: ${txHash}`)
     }
   }
-  const checkAndGrantUpdater = ({ classId, wallet }: { classId: string, wallet: string }) => {
+  const checkAndGrantUpdater = async ({ classId, wallet }: { classId: string, wallet: string }) => {
+    await checkWalletBalance({ wallet })
     return checkAndGrantRole({ classId, wallet }, 'UPDATER_ROLE')
   }
-  const checkAndGrantMinter = ({ classId, wallet }: { classId: string, wallet: string }) => {
+  const checkAndGrantMinter = async ({ classId, wallet }: { classId: string, wallet: string }) => {
+    await checkWalletBalance({ wallet })
     return checkAndGrantRole({ classId, wallet }, 'MINTER_ROLE')
   }
+  const checkWalletBalance = async ({ wallet }: { wallet: string }) => {
```

Comment:
user might not expect the function to throw error based on its name
```suggestion
  const assertPositiveWalletBalance = async ({ wallet }: { wallet: string }) => {
```